### PR TITLE
Update description of templates

### PR DESCRIPTION
**Pull request by jcantrill**

### CODE REVIEW

**Comment by jcantrill** [comment]
@bparees @soltysh @mfojtik Please review

File: using_openshift/templates.adoc [hunk 1]
```diff
@@ -9,4 +9,47 @@
 
 toc::[]
 
-== Overview
\ No newline at end of file
+== Overview
+A template describes a set of resources intended to be used together that can be customized and processed to produce a configuration.  Each
+template can define a list of environment variables that can be modified for consumption by containers.
```
Comment (bparees) [inline using_openshift/templates.adoc]: s/environment variables/parameters/ ?

File: using_openshift/templates.adoc [hunk 2]
```diff
@@ -9,4 +9,47 @@
...
+
+****
+`osc process -f [replaceable]#<filename>#`
+****
+
+Continuing the previous example, overriding the *ADMIN_USERNAME* and *MYSQL_DATABASE* produces a *json* string that can be applied to the server.
+The *json* string represents a *Config* with customized environment variables.
+
+----
+$osc process -f examples/sample-app/application-template-dockerbuild.json -v ADMIN_USERNAME=root,MYSQL_DATABASE=admin
```
Comment (bparees) [inline using_openshift/templates.adoc]: I would have gone for ADMIN_PASSWORD as the second parameter to override, myself.

**Comment by bparees** [comment]
Ultimately this probalby needs to provide more information, or point to more information, about the other sections beyond parameters (eg deploymentconfigs) but since the Config object is going away and being replaced with a List, such edits might be premature.

**Pull request merged**